Proper Support for ThinkPad X230 Tablet

late-in-autumn opened the pull request:
So I dug into the coreboot KConfig tonight and noticed that on the latest codebase, ThinkPad X230T is now an independent board, separate from the plain X230. So using the current X230 config as a starting point, by simply changing the board type to X230T had issue #145 completely eliminated (both the USB 2.0 cap and bootable device issue). I figured out that it would be beneficial to share some of my changes so that other X230T users could benefit too.

The x230t folder is basically a copy of the x230 folder with some minor modifications, and the code is probably not merge-ready yet since I haven't modified the util scripts thoroughly. However, I thought it would be better to open this pull request early, so I can make sure my changes are reviewed early, rather than late.

Sincerely

merge commented on x230t/NEWS:
```diff
@@ -0,0 +1,166 @@
+
```
don't copy over the NEWS file. create a new one. we want to do seperate skulls-x230t releases.

merge commented on x230t/README.md:
```diff
@@ -0,0 +1,272 @@
+# Skulls - [Thinkpad X230](https://pcsupport.lenovo.com/en/products/laptops-and-netbooks/thinkpad-x-series-laptops/thinkpad-x230) and X230T
+
```
update link to x230t only.

merge commented on x230t/release.sh:
```diff
@@ -0,0 +1,150 @@
+	exit 1
+fi
+
+RELEASE_DIR="skulls-x230-${version}"
```
it should be a "skulls-x230t-${version}" release.

(I'm sure I don't catch all needed changes now, I just want to point out the general idea)

merge commented:
Also, you don't include defconfig files, but the full config files. Use `make savedefconfig` which creates a "defconfig" file to use here.

Thanks a lot! It's a great start! Remember, if coreboot upstream differentiates the Tablet from the Laptop now, the defconfig change is the most important one. Add that and the mimum of scripts needed (nothing for Heads yet, for example) by 1:1 copy+pasting scripts in one commit and changing "x230 to x230t" in a second commit.

late-in-autumn commented:
Sorry, my Git history is probably a bit messy now, as I was a bit disorganized when starting this, but eventually figured out.

I did update the build and flash scripts, and I have currently tested the updated build, release, and internal flashing scripts (haven't tested the external flashing ones yet). Also updates some links and filenames within the README file. I am also wondering if we need to replace some of the images? I have a WLAN card upgrade on the way so I can snap a few pictures when doing that upgrade.

Changes regarding building coreboot itself are of the following:
- Generated proper defconfig files instead of the full config files
- Replaced the original VGA Option ROM blob with one that I extracted myself from my OEM BIOS backup

I have flashed both the free and non-free builds, and they both booted correctly. My new VGA Option ROM also seems to work correctly. Will boot into Windows later to confirm that.

I have an exam that's coming up, so it might be a few days before I can start working on this again.

merge commented:
It might take a bit longer these days to get around to review. I won't forget you though. Of couse anybody else reviewing helps too.



Am 6. April 2020 16:49:32 MESZ schrieb Sijie Bu <notifications@github.com>:
>Sorry, my Git history is probably a bit messy now, as I was a bit
>disorganized when starting this, but eventually figured out.
>
>I did update the build and flash scripts, and I have currently tested
>the updated build, release, and internal flashing scripts (haven't
>tested the external flashing ones yet). Also updates some links and
>filenames within the README file. I am also wondering if we need to
>replace some of the images? I have a WLAN card upgrade on the way so I
>can snap a few pictures when doing that upgrade.
>
>Changes regarding building coreboot itself are of the following:
>- Generated proper defconfig files instead of the full config files
>- Replaced the original VGA Option ROM blob with one that I extracted
>myself from my OEM BIOS backup
>
>I have flashed both the free and non-free builds, and they both booted
>correctly. My new VGA Option ROM also seems to work correctly. Will
>boot into Windows later to confirm that.
>
>I have an exam that's coming up, so it might be a few days before I can
>start working on this again.

-- 
Martin Kepplinger
xmpp: martink@jabber.at
sent from mobile

late-in-autumn commented:
Hope that all is well over there. Here in Los Angeles things are a bit messy too, so it will probably be a bit slow on my side too, and we are getting close to term finals.

merge commented:
as you've see I've created a PR for adding the X230T myself now too. it focuses on the essentials only. so, if you're ok with that, I'd appreciate if you could review over there: https://github.com/merge/skulls/pull/161 and the things you included in **this** PR, you could submit seperately:

* the vbios binary (if needed? what's different?)
* "Force container to use host network (NAT) " (explain why you want that)

thanks a lot for pushing this forward!

merge commented:
what would also be interesting: why did you change the configuration other than only changing the board type?

merge commented:
please make changes to the x230t config to what we currently have. thanks!
